index.js

```diff
@@ -77,14 +77,18 @@ export default function pTimeout(promise, options) {
 				return;
 			}
 
-			const errorMessage = typeof message === 'string' ? message : `Promise timed out after ${milliseconds} milliseconds`;
-			const timeoutError = message instanceof Error ? message : new TimeoutError(errorMessage);
-
 			if (typeof promise.cancel === 'function') {
 				promise.cancel();
 			}
 
-			reject(timeoutError);
+			if (message === false) {
+				resolve();
+			} else if (message instanceof Error) {
+				reject(message);
+			} else {
+				const errorMessage = message ?? `Promise timed out after ${milliseconds} milliseconds`;
+				reject(new TimeoutError(errorMessage));
```

Comment:
Is it worth exporting this helper? I find this pattern in every place where a customizable error is desired:


https://github.com/pixiebrix/pixiebrix-extension/blob/36a70a95a6979fa1145292ffc7624f296aa07fa0/src/utils/expectContext.ts#L25-L44

```js
reject(createError(`Promise timed out after ${milliseconds} milliseconds`, message))
```

Reply:
👍

Reply:
I haven't done that yet, I haven't thought of any API to specify both message and constructor. I'll publish it under the name [`as-error`](https://www.npmjs.com/package/as-error?activeTab=explore) unless you have better suggestions. All "create-*" and "cast-*" options were taken because until recently Error subclassing was not a thing.

Side note: I think specifying a whole error instance isn't ideal, what do you think? Maybe it should be:

```js
error: 'Custom Message' // uses default TimeoutError
// or
error: 'CustomConstructorError' // uses default message
// or
error: [CustomConstructorError, 'Custom Message'] // calls new CustomConstructorError('Custom Message')
```

The alternative would be to reinstantiate `error: new MyError('soup')`  via `new error.constructor(error.message)` but that might be unexpected?

Reply:
I'm considering releasing a more generic "error utils" package with https://github.com/refined-github/refined-github/blob/951f267db9441ae084ee660725d878bc1717bf5d/source/helpers/abort-controller.ts#L11 too